src/components/LayerCollection/LayerPOLine.js

```diff
@@ -30,16 +32,56 @@ class LayerPOLine extends Component {
 
   constructor(props) {
     super(props);
+
+    this.state = {
+      openModal: false,
+    };
     this.transitionToParams = transitionToParams.bind(this);
     this.connectedPOLineForm = props.stripes.connect(POLineForm);
   }
 
-  submitPOLine(data) {
+  openModal = () => {
+    this.setState({
+      openModal: true,
+    });
+  }
+
+  closeModal = () => {
+    this.setState({
+      openModal: false,
+    });
+  }
+
+  submitPOLine = async (data) => {
     const newLine = cloneDeep(data);
     const { lineMutator, onCancel } = this.props;
 
-    lineMutator.POST(newLine).then(() => {
+    try {
+      const response = await lineMutator.POST(newLine);
+      if (!response.id) throw new Error(response);
```

Comment:
Are you sure that object structure will be the same? I don;t think so. Please describe the purpose of this line